File: APP MOBILE/smart_money/lib/modules/transaction/models/report/financial_report_response.dart
// modules/transaction/models/report/financial_report_response.dart
// Map đúng với FinancialReportResponse.java của Spring Boot
// Tổng hợp cho màn hình Dashboard "All-in-One" (REPORT)

import 'transaction_report_response.dart';
import 'category_report_dto.dart';

class FinancialReportResponse {
  final TransactionReportResponse summary;     // Tổng quan tài chính trong kỳ
  final double totalCurrentBalance;             // Tổng tài sản hiện tại (tất cả ví)
  final List<CategoryReportDTO> expenseCategories; // Chi tiết biểu đồ Chi
  final List<CategoryReportDTO> incomeCategories;  // Chi tiết biểu đồ Thu

  const FinancialReportResponse({
    required this.summary,
    required this.totalCurrentBalance,
    required this.expenseCategories,
    required this.incomeCategories,
  });

  factory FinancialReportResponse.fromJson(Map<String, dynamic> json) {
    return FinancialReportResponse(
      summary: TransactionReportResponse.fromJson(
          json['summary'] as Map<String, dynamic>),
      totalCurrentBalance: (json['totalCurrentBalance'] as num).toDouble(),
      expenseCategories: (json['expenseCategories'] as List<dynamic>)
          .map((e) => CategoryReportDTO.fromJson(e as Map<String, dynamic>))
          .toList(),
      incomeCategories: (json['incomeCategories'] as List<dynamic>)
          .map((e) => CategoryReportDTO.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}

File: smartmoneyserver/src/main/java/fpt/aptech/server/dto/transaction/report/FinancialReportResponse.java
package fpt.aptech.server.dto.transaction.report;

import lombok.Builder;

import java.math.BigDecimal;
import java.util.List;

/**
 * DTO tổng hợp cho màn hình Dashboard "All-in-One".
 * Chứa tất cả thông tin cần thiết để hiển thị bức tranh tài chính toàn cảnh.
 */
@Builder
public record FinancialReportResponse(
    // 2. Tổng tài sản hiện tại của người dùng (Tất cả các ví)
    BigDecimal totalCurrentBalance,

    // 3. Danh sách chi tiết cho biểu đồ Chi phí (Kèm trung bình ngày)
    List<CategoryReportDTO> expenseCategories,

    // 4. Danh sách chi tiết cho biểu đồ Thu nhập (Kèm trung bình ngày)
    List<CategoryReportDTO> incomeCategories
) {}
